### DISCUSSION TITLE
Unable to rebuild APK using apktool 2.9.3

### DISCUSSION THREAD

**Discussion by Meenabalveer95** [iBotPeaches/Apktool#3669] [Q&A] [answered]
apktool b study -o study.apk
I: Using Apktool 2.9.3
I: Checking whether sources has changed...
I: Checking whether resources has changed...
I: Building resources...
brut.androlib.exceptions.AndrolibException: brut.common.BrutException: could not exec (exit code = 2): [/tmp/brut_util_Jar_164566533652547646923478157524024214531.tmp, compile, --dir, /sdcard/Download/study.com/res, --legacy, -o, /sdcard/Download/study.com/build/resources.zip]
W: /tmp/brut_util_Jar_164566533652547646923478157524024214531.tmp: 4: Syntax error: Unterminated quoted string

**Comment by iBotPeaches** [comment, marked as answer]
Apktool is not supported on device. I can tell with your stacktrace you are attempting on an sdcard which suggests an Android device.

**Comment by Meenabalveer95** [comment]
apktool b '/home/kali/Desktop/study' -o study.apk
I: Using Apktool 2.9.3
I: Checking whether sources has changed...
I: Smaling smali folder into classes.dex...
I: Checking whether sources has changed...
I: Smaling smali_classes10 folder into classes10.dex...
I: Checking whether sources has changed...
I: Smaling smali_classes2 folder into classes2.dex...
I: Checking whether sources has changed...
I: Smaling smali_classes3 folder into classes3.dex...
I: Checking whether sources has changed...
I: Smaling smali_classes4 folder into classes4.dex...
I: Checking whether sources has changed...
I: Smaling smali_classes5 folder into classes5.dex...
I: Checking whether sources has changed...
I: Smaling smali_classes6 folder into classes6.dex...
I: Checking whether sources has changed...
I: Smaling smali_classes7 folder into classes7.dex...
I: Checking whether sources has changed...
I: Smaling smali_classes8 folder into classes8.dex...
I: Checking whether sources has changed...
I: Smaling smali_classes9 folder into classes9.dex...
I: Checking whether resources has changed...
I: Building resources...
brut.androlib.exceptions.AndrolibException: brut.common.BrutException:
could not exec (exit code = 2):
[/tmp/brut_util_Jar_85324216882112025259153750803802641963.tmp, compile,
--dir, /home/kali/Desktop/study/res, --legacy, -o,
/home/kali/Desktop/study/build/resources.zip]
W: /tmp/brut_util_Jar_85324216882112025259153750803802641963.tmp: 4: Syntax
error: Unterminated quoted string
